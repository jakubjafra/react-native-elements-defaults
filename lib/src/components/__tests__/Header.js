const testHelperUtils = require('../../testHelperUtils');

const utils = require('../../utils');
const Overwrite = utils.Overwrite;

const name = 'Header',
    __oldElement__ = require('react-native-elements/src/header/Header'),
    __element__ = require('../Header'),
    noTheme = new Overwrite(
        name,
        Overwrite.TYPE.FUNCTIONAL_COMPONENT_ATTRIBUTES,
        Overwrite.SCOPE.DEFAULTS,
        {}
    ),
    basicTheme = new Overwrite(
        name,
        Overwrite.TYPE.FUNCTIONAL_COMPONENT_ATTRIBUTES,
        Overwrite.SCOPE.DEFAULTS,
        {
            containerStyle: {
                backgroundColor: 'red',
            }
        }
    );

describe(name, () => {
    testHelperUtils.defaultTests(name, __oldElement__, __element__, noTheme, basicTheme);
});
