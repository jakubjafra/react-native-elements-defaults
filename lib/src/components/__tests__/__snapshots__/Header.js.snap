// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with basic theme applied Header renders correctly 1`] = `
<View
  containerStyle={
    Object {
      "backgroundColor": "red",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "#476DC5",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 1,
        "height": 70,
        "padding": 15,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Header with basic theme applied OldHeader renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#476DC5",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 1,
        "height": 70,
        "padding": 15,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Header with no theme applied Header renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#476DC5",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 1,
        "height": 70,
        "padding": 15,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Header with no theme applied OldHeader and Header returns the same thing if no theme is applied 1`] = `"[2mCompared values have no visual difference.[22m"`;

exports[`Header with no theme applied OldHeader renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#476DC5",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 1,
        "height": 70,
        "padding": 15,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  />
</View>
`;
